osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -275,6 +276,30 @@ public async Task ChangeState(MultiplayerUserState newState)
             }
         }
 
+        public async Task ChangeBeatmapAvailability(BeatmapAvailability newBeatmapAvailability)
+        {
+            using (var userUsage = await GetOrCreateLocalUserState())
+            using (var roomUsage = await getLocalUserRoom(userUsage.Item))
+            {
+                var room = roomUsage.Item;
+
+                if (room == null)
+                    throw new InvalidStateException("Attempted to operate on a null room");
```

Comment:
another case of incorrect cleanup (should be resolved via #41 so not necessarily needing changes here).

Reply:
This should be an `InvalidOperationException`. I'm not sure why you made it an exception which will be returned to the user (this is not how it's done anywhere else).